src/ElmFormat/Render/Box.hs

```diff
@@ -1493,51 +1494,57 @@ formatExpression' elmVersion importInfo context aexpr =
                         ]
                     |> expressionParens AmbiguousEnd context -- TODO: not tested
 
-        AST.Expression.Case (subject,multiline) clauses ->
+        AST.Expression.Case (subject, multilineSubject) clauses multilineClauses ->
             let
                 opening =
                   case
-                    ( multiline
+                    ( multilineSubject
                     , formatCommentedExpression elmVersion importInfo SyntaxSeparated subject
                     )
                   of
                       (False, SingleLine subject') ->
-                          line $ row
-                              [ keyword "case"
-                              , space
-                              , subject'
-                              , space
-                              , keyword "of"
-                              ]
+                          line $ row [ keyword "case" , space , subject' , space , keyword "of" ]
                       (_, subject') ->
                           stack1
                               [ line $ keyword "case"
                               , indent subject'
                               , line $ keyword "of"
                               ]
 
-                clause (pat, expr) =
+                clausesBodies =
+                    map (formatHeadCommentedStack (formatExpression elmVersion importInfo SyntaxSeparated) . snd) clauses
+
```

Comment:
Maybe I should use `mapAccumR` instead of `map` in `clause` and `foldl'` in `splitAllClauses`.